### PR TITLE
fix(menu): Styling active menu in SPA navigation

### PR DESCRIPTION
**Pull request by justinpark**
### SUMMARY
Fixes the addressed issue (no. 2) on https://github.com/apache/superset/pull/25151#issuecomment-1747279563

This commit replaces Link with NavLink to fix outdated highlighted menu. This commit replaces the ant-menu-item-selected style by NavLink active style to sync the router active link status.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Before:

https://github.com/apache/superset/assets/1392866/5a4e20a0-e393-4d63-98ad-7789a44a4cce

After:

https://github.com/apache/superset/assets/1392866/23cdd7ea-aed0-44fa-94d2-33dc08c0a03f

### TESTING INSTRUCTIONS
Click Chart (or Dashboard) link
Click Query History menu under the SQL Link
Navigate back by the browser button
See the highlighted menu on the header

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by EugeneTorap** [comment]
Hi @michael-s-molina @rusackas! Can we merge it?

**Comment by EugeneTorap** [comment]
Hi @michael-s-molina @john-bodley @craig-rueda! Can you approve this PR?

File: superset-frontend/src/features/home/Menu.tsx [hunk 1]
```diff
@@ -154,6 +154,29 @@ const globalStyles = (theme: SupersetTheme) => css`
       margin-left: ${theme.gridUnit * 1.75}px;
     }
   }
+  .ant-menu-item-selected {
+    background-color: transparent !important;
```
Comment (michael-s-molina) [inline superset-frontend/src/features/home/Menu.tsx:158]: Is it possible to avoid `!important` here? Maybe using a more specific class?

Comment (justinpark) [reply superset-frontend/src/features/home/Menu.tsx]: @michael-s-molina I just removed all these redundant `!important`

**Review by michael-s-molina — APPROVED** [review]
LGTM

**Pull request merged**